pkg/notify/notify.go

```diff
@@ -103,6 +104,28 @@ func (r *Receiver) Notify(data *alertmanager.Data, hashJiraLabel bool, updateSum
 			}
 		}
 
+		if r.conf.UpdateInComment.Valid && r.conf.UpdateInComment.Bool {
+			numComments := 0
+			if issue.Fields.Comments != nil {
+				numComments = len(issue.Fields.Comments.Comments)
+			}
+			if numComments > 0 && issue.Fields.Comments.Comments[(numComments-1)].Body == issueDesc {
+				// if the new comment is identical to the most recent comment,
+				// this is probably due to the prometheus repeat_interval and should not be added.
+				level.Debug(r.logger).Log("msg", "not adding new comment identical to last", "key", issue.Key)
+			} else if numComments == 0 && issue.Fields.Description == issueDesc {
+				// if the first comment is identical to the description,
+				// this is probably due to the prometheus repeat_interval and should not be added.
+				level.Debug(r.logger).Log("msg", "not adding comment identical to description", "key", issue.Key)
+			} else {
+				retry, err := r.addComment(issue.Key, issueDesc)
+				if err != nil {
+					return retry, err
+				}
+			}
+		}
+
+		// update description after possibly adding a comment so that it's possible to detect redundant first comment
```

Comment:
I would remove the comment - users can turn updateDescription off if they want (via flag)

NOTE: Not sure why some options are in flag, some in config --sounds like we could move all of them to config one day (:

Reply:
That comment is there to warn against updating the description **before** adding a comment (in terms of order-of-oppertions), because then we wouldn't be able to compare what the description used to be. This is needed when there are 0 comments and we are trying to determine if the first comment will be a duplicate of the description and should not be added.

Reply:
Ok, thanks. Then I would propose slight language change to the comment as `so that it's possible to detect redundant first comment` might be vague.

```suggestion
		// update description if enabled. This has to be done after comment adding logic which needs to handle redundant commentary vs description case.
```